### PR TITLE
Fixes NPE in /node-state REST calls when the Node is not fully activated

### LINKED ISSUE #17773: NPE when retrieving node-state before the node becomes ACTIVE
This issue was reported by Even Holthe on Slack:

I’m trying to set it up via the helm charts on Azure Kubernetes Service. However, on startup I get the following cryptic error:
```
Oct 27, 2020 10:36:54 AM com.hazelcast.internal.ascii.rest.HttpPostCommandProcessor
WARNING: [10.20.16.38]:5701 [dev] [4.0.3] An error occurred while handling request HttpCommand [HTTP_GET]{uri='/hazelcast/health/node-state'}AbstractTextCommand[HTTP_GET]{requestId=0}
java.lang.NullPointerException
	at com.hazelcast.internal.ascii.rest.HttpGetCommandProcessor.handleHealthcheck(HttpGetCommandProcessor.java:137)
	at com.hazelcast.internal.ascii.rest.HttpGetCommandProcessor.handle(HttpGetCommandProcessor.java:79)
	at com.hazelcast.internal.ascii.rest.HttpGetCommandProcessor.handle(HttpGetCommandProcessor.java:47)
	at com.hazelcast.internal.ascii.TextCommandServiceImpl$CommandExecutor.run(TextCommandServiceImpl.java:396)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```
I’m seeing a lot of people getting it to run successfully on Kubernetes, so I’m thinking this must be a problem with my cluster somehow. Since this seems to be an internal Hazelcast issue, I’m unsure whether my network policy configuration/RBAC config can be blamed. Has anyone been in the same situation or seen similar issues? Any pointers would be greatly appreciated :crossed_fingers:

### PR DESCRIPTION
**Pull request by kwart**
Fixes #17773 .

### CODE REVIEW

**Comment by evenh** [comment]
Oh, that was quick! I was stuck in meetings but I see that the details is correct 👌🏻

**Comment by evenh** [comment]
Would it be possible to backport/merge this fix into 4.1 as well?

**Comment by mmedenjak** [comment]
@evenh yes, I believe we can, after this fix has been merged

**Review by vbekiaris — APPROVED** [review]

File: hazelcast/src/test/java/com/hazelcast/internal/ascii/RestNodeStateTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        Config config = smallInstanceConfig().setProperty(ClusterProperty.DISCOVERY_SPI_ENABLED.getName(), "true");
+        RestApiConfig restApiConfig = new RestApiConfig().setEnabled(true).enableAllGroups();
+        NetworkConfig networkConfig = config.getNetworkConfig();
+        networkConfig.setPort(5000).setPortAutoIncrement(false);
+        networkConfig.getJoin().getMulticastConfig().setEnabled(false);
+        networkConfig.setRestApiConfig(restApiConfig);
+        StrategyFactory discoveryStrategyFactory = new StrategyFactory();
+        networkConfig.getJoin().getDiscoveryConfig()
+                .addDiscoveryStrategyConfig(new DiscoveryStrategyConfig(discoveryStrategyFactory));
+        Executors.newSingleThreadExecutor().execute(() -> Hazelcast.newHazelcastInstance(config));
```
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/internal/ascii/RestNodeStateTest.java:76]: nit: could also use `HazelcastTestSupport#spawn(Runnable)` here

**Review by mmedenjak — APPROVED** [review]

**Pull request merged**